core/src/validator.rs

```diff
@@ -563,8 +568,71 @@ impl Validator {
     }
 }
 
+fn active_vote_account_exists_in_bank(bank: &Arc<Bank>, vote_account: &Pubkey) -> bool {
+    if let Some(account) = &bank.get_account(vote_account) {
+        if let Some(vote_state) = VoteState::from(&account) {
+            return !vote_state.votes.is_empty();
+        }
+    }
+    false
+}
+
+fn post_process_restored_tower(
+    restored_tower: crate::consensus::Result<Tower>,
+    validator_identity: &Pubkey,
+    vote_account: &Pubkey,
+    config: &ValidatorConfig,
+    ledger_path: &Path,
+    bank_forks: &BankForks,
+) -> Tower {
+    restored_tower
+        .and_then(|tower| {
+            let root_bank = bank_forks.root_bank();
+            let slot_history = root_bank.get_slot_history();
+            tower.adjust_lockouts_after_replay(root_bank.slot(), &slot_history)
+        })
+        .unwrap_or_else(|err| {
+            let is_already_active =
+                active_vote_account_exists_in_bank(&bank_forks.working_bank(), &vote_account);
+            if config.require_tower && is_already_active {
+                error!("Requested mandatory tower restore failed: {:?}", err);
+                error!(
+                    "And there is an existing vote_account containing actual votes. \
+                       Aborting due to possible conflicting duplicate votes"
+                );
+                process::exit(1);
+            }
+            let not_found = if let TowerError::IOError(io_err) = &err {
+                io_err.kind() == std::io::ErrorKind::NotFound
```

Comment:
Hehe once this is checked in, we might want to communicate with validators to take extra precautions with this tower. People's disks have been failing left and right these days :)

Reply:
Yeah, this is true.